### PR TITLE
implement poc configuration of IHD/HF with transient states

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: proof-of-concept
- *JIRA issue*: [MIC-4056](https://jira.ihme.washington.edu/browse/MIC-4056)
- *Research reference*: 

### Changes and notes
Added support for transient states
Added support for custom get_data_functions
Updated the IHD/HF model to incorporate transient states for exiting susceptible and post-mi
Refactored existing disease model to allow the get data functions to be referenced
Added special DiseaseObserver to account for transient state
Updated results processing to count IHD/HF data

### Verification and Testing
Ran full parallel simulation and ran Syl's validation notebook against it

### CODE REVIEW

File: src/vivarium_nih_us_cvd/components/observers.py [hunk 1]
```diff
@@ -334,3 +336,30 @@ def on_time_step_prepare(self, event: Event):
     def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
         metrics.update(self.counts)
         return metrics
+
+
+class TransientIHDAndHFObserver(DiseaseObserver_):
+    def _get_transitions(self, builder: Builder) -> List[TransitionString]:
+        transitions = [
+            TransitionString(
+                "susceptible_to_ischemic_heart_disease_and_heart_failure_TO_acute_myocardial_infarction"
```
Comment (stevebachmeier) [inline src/vivarium_nih_us_cvd/components/observers.py:345]: nit: "TO"

Comment (stevebachmeier) [reply src/vivarium_nih_us_cvd/components/observers.py:345]: Or is that how we differentiate a transitional "to"?

File: src/vivarium_nih_us_cvd/configuration/causes.yaml [hunk 2]
```diff
@@ -79,40 +86,52 @@ causes:
                     disability_weight: "cause.acute_myocardial_infarction.disability_weight"
                     excess_mortality_rate: "cause.acute_myocardial_infarction.excess_mortality_rate"
         transitions:
-            acute_mi_incidence:
+            susceptible_transient_incidence:
                 source: "susceptible"
-                sink: "acute_myocardial_infarction"
+                sink: "transient_susceptible_state"
```
Comment (stevebachmeier) [inline src/vivarium_nih_us_cvd/configuration/causes.yaml:93]: is this effectively defining a self-transition?

**Review by stevebachmeier — APPROVED** [review]

File: src/vivarium_nih_us_cvd/components/observers.py (hunk 1 above)
Comment (hussain-jafari) [reply src/vivarium_nih_us_cvd/components/observers.py:345]: Yeah the capital TO allows the Transition String to properly parse the string and get the from state and to state.

File: src/vivarium_nih_us_cvd/configuration/causes.yaml (hunk 2 above)
Comment (hussain-jafari) [reply src/vivarium_nih_us_cvd/configuration/causes.yaml:93]: The transient state (called the transition state in the link) is described [here](https://vivarium-research.readthedocs.io/en/latest/models/causes/ischemic_heart_disease/gbd_2019/index.html#mi-and-hf).

**Review by mattkappel — APPROVED** [review]

**Pull request merged**